### PR TITLE
[Fix] URL maximum length in E-Document Integration

### LINKED ISSUE #26672: [Bug]: URL maximum length in E-Document Integration
### Describe the issue

When implementing the interface Microsoft.eServices.EDocument."E-Document Integration" to Send E-Documents we came across the issue that request URIs which are longer than 250 characters lead to an error in procedure InsertIntegrationLog of Codeunit 6132 "E-Document Log" due to the validation of the URL record field 7 of table 6127 "E-Document Integration Log" which is limited to 250 characters.

### Expected behavior

The supported maximum length of the URL should be longer.

### Steps to reproduce

1. Create a Codeunit which implements "E-Document Integration"
2. Implement procedure "Send" using a request URI with more than 250 characters in parameter HttpRequest
3. Error will be raised by the field validation in the subsequent call to the procedure InsertLog in codeunit 6132 "E-Document Log" due to the length of the RequestUri which is longer than 250 characters.

image

### Additional context

The .NET System.Uri class limits the maximum length of a URI for the System.Net.Http.HttpClient to [65519 characters](https://learn.microsoft.com/en-us/dotnet/api/system.uri.-ctor?view=net-8.0#system-uri-ctor(system-string)). Per [RFC 9110](https://www.rfc-editor.org/rfc/rfc9110#section-4.1-5) (published in June 2022) there is no fixed limit for URIs, but it is recommended to support, "at a minimum, URIs with lengths of 8000 octets in protocol elements". Given the Business Central context, we would argue for URIs limited to not more than 2048 characters as an expected limit.

As a workaround we currently update the request URI of the HttpRequest parameter before exiting the procedure as follows:

image

### I will provide a fix for a bug

- [X] I will provide a fix for a bug

### PR DESCRIPTION
**Pull request by pluethi**
#### Summary 
Added a new field "Request URL" to table 6127 "E-Document Integration Log" with a maximum string length of 2048 to allow for longer URLs when using the "E-Document Integration" interface. The new filed replaces the existing field URL. I marked the existing field as obsolete and updated three (all) references from the old URL field to the new "Request URL" field.

#### Work Item(s) 
Fixes #26672

Fixes [AB#540448](https://dynamicssmb2.visualstudio.com/1fcb79e7-ab07-432a-a3c6-6cf5a88ba4a5/_workitems/edit/540448)

### CODE REVIEW

**Comment by pluethi** [comment]
@microsoft-github-policy-service agree

File: Apps/W1/EDocument/app/src/Log/EDocumentIntegrationLogs.Page.al [hunk 1]
```diff
@@ -31,7 +31,7 @@ page 6128 "E-Document Integration Logs"
                     {
                         ToolTip = 'Specifies the service code for the document.';
                     }
-                    field(URL; Rec.URL)
+                    field(URL; Rec."Request URL")
```
Comment (JesperSchulz) [inline Apps/W1/EDocument/app/src/Log/EDocumentIntegrationLogs.Page.al:34]: Given we now reference the new field, shouldn't we have upgrade code then? Using DataTransfer that should be a swift procedure.

Comment (pluethi) [reply Apps/W1/EDocument/app/src/Log/EDocumentIntegrationLogs.Page.al:34]: That's true, of course. I had not considered this, as it is "only" a log. However, it is visible to the user, so as a user I would also expect the existing data to be transferred. I will add this.

**Comment by JesperSchulz** [comment]
Upgrade code added internally. Will merge once it has passed the build system.

**Review by JesperSchulz — APPROVED** [review]
I've added upgrade code during merge to local repo.

File: Apps/W1/EDocument/app/src/Log/EDocumentIntegrationLogs.Page.al (hunk 1 above)
Comment (JesperSchulz) [reply Apps/W1/EDocument/app/src/Log/EDocumentIntegrationLogs.Page.al:34]: I've added the upgrade code internally. Case closed.

**Comment by JesperSchulz** [comment]
Thanks for reporting this. We agree, and we’ll publish a fix asap, either in an update for the current version or in the next major release. Please do not reply to this, as we do not monitor closed issues. If you have follow-up questions or requests, please create a new issue where you reference this one.

Build ID: 22963.

**Pull request merged**